### PR TITLE
91-AddCounterCartNavBar :: Added cart counter and showing in navbar

### PR DESCRIPTION
**Pull request by daynewright**
## Status

**READY/IN DEVELOPMENT**
## Documentation

[ ] I have fully documented features in this PR's code. Please put an `X` in the box to confirm.

[ ] This PR does **NOT** require documentation.
## Description

A few sentences describing the overall goals of the pull request's commits.

Calculates the line item count of products and adds them to base model.  Displays it on by the cart icon in the navbar.
## Resolves Issue Number
#91
## Deploy Notes

Notes regarding deployment the contained body of work.  These should note any
db migrations, etc.

``` sh
git pull --prune
git checkout 91-AddCounterCartNavBar
dotnet run
```
## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.
1.  dotnet run
2.  select user

**IF (order exists)**
3. cart icon should show a counter of items on the order

**ELSE**
3. select product
4. add to cart
5. counter should show 1
## Impacted Files in Application

List general components of the application that this PR will affect:
- Controllers/CartController.cs
- ViewModels/Base.cs
- Views/Shared/_NavbarView.cshtml

### CODE REVIEW

File: ViewModels/Base.cs [hunk 1]
```diff
@@ -54,6 +56,18 @@ public BaseViewModel(BangazonContext ctx)
                 Text = $"{li.FirstName} {li.LastName}",
                 Value = li.UserId.ToString()
             });
+
+        var CartProducts = (
```
Comment (MattKraatz) [inline ViewModels/Base.cs]: explicit declaration please, I think:
`List<Product> CartProducts = (`

**Review by MattKraatz — CHANGES_REQUESTED** [review]
LGTM after a quick style guide change

**Review by Anulfo — COMMENTED** [review]
LGTM

File: ViewModels/Base.cs (hunk 1 above)
Comment (daynewright) [reply ViewModels/Base.cs]: @MattKraatz  I changed it to add the `CartProducts` in the constructor on the base.  I removed the LINQ query from the cart controller and just use this instance of it in that cart view.  Seemed to be more DRY.  Also added comments.  All that to say you might want to review the changes and test again.

**Review by MattKraatz — APPROVED** [review]

**Comment by MattKraatz** [comment]
LGTM

**Comment by garrett-vangilder** [comment]
LGTM

**Pull request merged**